source/apiVolontaria/apiVolontaria/views.py

```diff
@@ -257,3 +265,144 @@ def post(self, request):
                 {'detail': error},
                 status=status.HTTP_500_INTERNAL_SERVER_ERROR
             )
+
+
+class ResetPassword(APIView):
+    """
+    post:
+    Create a new token allowing user to change his password.
+    """
+    permission_classes = ()
+    authentication_classes = ()
+
+    def post(self, request, *args, **kwargs):
+        if settings.CONSTANT['EMAIL_SERVICE'] is not True:
+            # Without email this functionality is not provided
+            return Response(status=status.HTTP_501_NOT_IMPLEMENTED)
+
+        # Valid params
+        serializer = serializers.ResetPasswordSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        # get user from the username given in data
+        try:
+            user = User.objects.get(username=request.data["username"])
+        except Exception:
+            content = {
+                'detail': _("No account with this username."),
+            }
+            return Response(content, status=status.HTTP_400_BAD_REQUEST)
+
+        # remove old tokens to change password
+        tokens = ActionToken.objects.filter(
+            type='password_change',
+            user=user,
+        )
+
+        for token in tokens:
+            token.expire()
+
+        # create the new token
+        token = ActionToken.objects.create(
+            type='password_change',
+            user=user,
+        )
+
+        # Send the new token by e-mail to the user
+        MAIL_SERVICE = settings.SETTINGS_IMAILING
+        FRONTEND_SETTINGS = settings.CONSTANT['FRONTEND_INTEGRATION']
+
+        button_url = FRONTEND_SETTINGS['FORGOT_PASSWORD_URL'].replace(
+            "{{token}}",
+            str(token)
+        )
+
+        email = IMailing.create_instance(
+            MAIL_SERVICE["SERVICE"],
+            MAIL_SERVICE["API_KEY"],
+        )
+
+        response_send_mail = email.send_templated_email(
+            email_from=MAIL_SERVICE["EMAIL_FROM"],
+            template_id=MAIL_SERVICE["TEMPLATES"]["FORGOT_PASSWORD"],
+            list_to=[user.email],
+            context={
+                "forgot_password_url": button_url,
+            },
+        )
+
+        if response_send_mail["code"] == "failure":
+            content = {
+                'detail': _("Your token has been created but no email "
+                            "has been sent. Please contact the "
+                            "administration."),
+            }
+            return Response(content, status=status.HTTP_201_CREATED)
+
+        else:
+            return Response(status=status.HTTP_201_CREATED)
```

Comment:
Would it be a good idea to send some content here.  

Could only be something like content = { 'detail': {_('Email succesfully sended')} }

Reply:
It can be done but for i wanted to have the same logic as the other call of the API.
Actually we return data only when we have a problem.